waku/v2/node/storage/message/waku_message_store.nim

```diff
@@ -31,13 +31,14 @@ proc init*(T: type WakuMessageStore, db: SqliteDatabase): MessageStoreResult[T]
 
   let prepare = db.prepareStmt("""
     CREATE TABLE IF NOT EXISTS """ & TABLE_TITLE & """ (
-        id BLOB PRIMARY KEY,
+        id BLOB,
         receiverTimestamp """ & TIMESTAMP_TABLE_TYPE & """ NOT NULL,
         contentTopic BLOB NOT NULL,
         pubsubTopic BLOB NOT NULL,
         payload BLOB,
         version INTEGER NOT NULL,
-        senderTimestamp """ & TIMESTAMP_TABLE_TYPE & """  NOT NULL
+        senderTimestamp """ & TIMESTAMP_TABLE_TYPE & """  NOT NULL,
+        CONSTRAINT messageIndex PRIMARY KEY (senderTimestamp, receiverTimestamp, id, pubsubTopic)
```

Comment:
Right, very good point. The primary key indeed acts as a data integrity constraint that would cause DB operations to fail with an exception when violated. This is expensive and should be avoided. In other words, we should deduplicate messages _before_ attempting to store them in the DB. This is what I've now done with [this return](https://github.com/status-im/nim-waku/pull/864/commits/d29d0e44eb3aca5dbfabde343ee8d4e7ac4a1481#diff-c574072f2cd8c3e7a6fbc74ba4aa947f6296e0fe18d455053f45b23447afcdb7R399). Of course, then there's also no need for the DB scheme to be permissive of duplicates, so I've also removed the `receiverTimestamp` from the PRIMARY KEY in [ed09cce](https://github.com/status-im/nim-waku/pull/864/commits/ed09cce6b890ba5ff13486d1f85df0b7c36db7e6). Thanks for catching this. 👍